src/DotNet/Resources/ResourceWriter.cs

```diff
@@ -78,20 +83,12 @@ public sealed class ResourceWriter {
 			writer.Write(dataStream.ToArray());
 		}
 
-		void WriteData(BinaryWriter writer, ResourceElement info, IFormatter formatter) {
+		void WriteData(ResourceBinaryWriter writer, ResourceElement info, IFormatter formatter) {
 			var code = GetResourceType(info.ResourceData);
-			WriteUInt32(writer, (uint)code);
+			writer.Write7BitEncodedInt((int)code);
```

Comment:
Is this identical to the original code?

Reply:
Yes, the new method is implemented as follows in the core library:
```csharp
protected void Write7BitEncodedInt(int value)
{
	uint num;
	for (num = (uint) value; num >= 128U; num >>= 7)
		this.Write((byte) (num | 128U));
	this.Write((byte) num);
}
```

This new method is called with the enum casted to `int` is equivalent to the previous implementation which casts the enum to `uint` because the core library method casts the `int` to a `uint` itself.
```csharp
static void WriteUInt32(BinaryWriter writer, uint value) {
	while (value >= 0x80) {
		writer.Write((byte)(value | 0x80));
		value >>= 7;
	}
	writer.Write((byte)value);
}
```